core/lib/zksync_core/src/sync_layer/snapshots.rs

```diff
@@ -0,0 +1,285 @@
+
+    if applied_snapshot_status.is_none() {
+        applied_snapshot_status = Some(AppliedSnapshotStatus {
+            l1_batch_number: l1_batch_number,
+            is_finished: false,
+            last_finished_chunk_id: None,
+        });
+    };
+    let mut applied_snapshot_status = applied_snapshot_status.unwrap();
+
+    storage
+        .applied_snapshot_status_dal()
+        .set_applied_snapshot_status(&applied_snapshot_status)
+        .await
+        .unwrap();
+
+    let recovered_version = l1_batch_number.0 as u64;
+    let db = RocksDB::new(Path::new(&merkle_tree_db_path));
+    let db: &mut dyn PruneDatabase = &mut (RocksDBWrapper::from(db));
+    let mut recovery = MerkleTreeRecovery::new(db, recovered_version);
+
+    sync_miniblock_header(storage, client, miniblock_number).await;
+
+    sync_l1_batch_metadata(storage, &snapshot.last_l1_batch_with_metadata).await;
+
+    applied_snapshot_status.is_finished = true;
+    storage
+        .applied_snapshot_status_dal()
+        .set_applied_snapshot_status(&applied_snapshot_status)
+        .await
+        .unwrap();
+
+    let blob_store = ObjectStoreFactory::snapshots_from_env()
+        .context("ObjectStoreFactor::snapshots_from_env()")?
+        .create_store()
+        .await;
+
+    for chunk in snapshot.chunks.iter() {
+        let storage_key = chunk.key;
+
+        let chunk_id = storage_key.chunk_id;
+        if applied_snapshot_status.last_finished_chunk_id.is_some()
+            && chunk_id > applied_snapshot_status.last_finished_chunk_id.unwrap()
+        {
+            tracing::info!(
+                "Skipping processing chunk {}, file already processed",
+                chunk_id
+            );
+        }
+        tracing::info!(
+            "Processing chunk {} located in {}",
+            chunk_id,
+            &chunk.filepath
+        );
+
+        let storage_snapshot_chunk: SnapshotChunk = blob_store.get(storage_key).await.unwrap();
+
+        let factory_deps = storage_snapshot_chunk.factory_deps;
+        sync_factory_deps_chunk(storage, miniblock_number, factory_deps).await;
+
+        let storage_logs = &storage_snapshot_chunk.storage_logs;
+        sync_storage_logs_chunk(storage, miniblock_number, storage_logs).await;
+
+        sync_initial_writes_chunk(storage, l1_batch_number, storage_logs).await;
+
+        sync_tree_chunk(&mut recovery, storage_logs).await;
+
+        applied_snapshot_status.last_finished_chunk_id = Some(chunk_id);
+        storage
+            .applied_snapshot_status_dal()
+            .set_applied_snapshot_status(&applied_snapshot_status)
+            .await
+            .unwrap();
+    }
+
+    tracing::info!("Processing chunks finished, finalizing merkle tree");
+
+    let tree = recovery.finalize();
+    let started_at = Instant::now();
+    tree.verify_consistency(recovered_version).unwrap();
```

Comment:
I think consistency verification should be opt-in or not performed at all. If local tests are any indication, it can take more time than the tree recovery.

Reply:
I changed so it's not performed at all